src/test/java/uk/gov/register/functional/AnalyticsFunctionalTest.java

```diff
@@ -38,56 +36,65 @@
         testEntry2 = TestEntry.anEntry(2, "{\"street\":\"" + testEntry2Key + "\",\"address\":\"12346\"}", "12346");
     }
 
-    @Rule
-    public RegisterRule register;
+    @ClassRule
+    public static RegisterRule register = new RegisterRule();
 
     @Before
     public void setup() {
         register.wipe();
-        register.mintLines(testEntry1.itemJson,
-                testEntry2.itemJson);
+        register.mintLines("address", testEntry1.itemJson, testEntry2.itemJson);
     }
 
     private final String targetUrl;
-    private final Boolean shouldIncludeAnalytics;
 
-    public AnalyticsFunctionalTest(String targetUrl, String trackingId, Boolean shouldIncludeAnalytics) {
-        this.register = createRegister(trackingId);
+    public AnalyticsFunctionalTest(String targetUrl) {
```

Comment:
I'll extract the private method.

You say "not clearly visible in the test" but I created the constants `REGISTER_WITH_MISSING_TRACKING_ID` etc specifically to document these assumptions. Is that not enough? Can you suggest anything more?